python_modules/libraries/dagster-dbt/dagster_dbt/core/resource.py

```diff
@@ -758,6 +759,14 @@ def _get_subset_selection_for_context(
     is_checks_subset = (
         assets_def.check_specs_by_output_name != assets_def.node_check_specs_by_output_name
     )
+    latest_code_version_by_key = context.instance.get_latest_materialization_code_versions(assets_def.keys)
```

Comment:
one thing to flag is that this data fetch could be expensive, particularly if there are many asset keys at play

Reply:
Good call out! I moved where this call occurs so I only use the asset keys that align with dbt view models.